[Chow Yew Meng Jonas] iP

jonas-chow opened the pull request:
# DukePro
_copied wholesale from the [week 4 project page](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html)_

---
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jonas-chow committed:
Add binary files to .gitignore

jonas-chow committed:
Add Greet, Echo, Exit functionality

jonas-chow committed:
Add Add, List functionality

jonas-chow committed:
Add Mark as Done functionality

jonas-chow committed:
Add ToDo, Event, Deadline functionality

jonas-chow committed:
Add Automated Text UI Testing functionality

jonas-chow committed:
Add Handle Errors functionality

jonas-chow committed:
Add Delete functionality, updated testcases

jonas-chow committed:
Added Use Enums functionality

jonas-chow committed:
Add Save functionality

jonas-chow committed:
Add Dates and Times functionality

jonas-chow committed:
Merge branch 'branch-Level-7'

jonas-chow committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/TaskList.java

jonas-chow committed:
Fix bugs from merging

jonas-chow committed:
Add More OOP functionality

jonas-chow committed:
Add Packages functionality

jonas-chow committed:
Add JUnit tests

jonas-chow committed:
Add Jar functionality

jonas-chow committed:
Add javadocs

jonas-chow committed:
Follow Coding Standard

jonas-chow committed:
Add Find functionality

jonas-chow committed:
Merge branch 'branch-A-JavaDoc'

jonas-chow committed:
Merge branch 'branch-A-CodingStandard'

jonas-chow committed:
Merge branch 'branch-Level-9'

jonas-chow committed:
Fix bug brought over from renaming

jonas-chow committed:
Follow coding standard

jonas-chow committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

jonas-chow committed:
Enforce coding standard with checkstyle

jonas-chow committed:
Add GUI functionality

jonas-chow committed:
Merge branch 'javafx-test'

eugenecsa commented on src/main/java/duke/application/Launcher.java:
```diff
@@ -0,0 +1,16 @@
+ */
+public class Launcher {
+    /**
+     * Launches the application
```
Remember to have an empty line between the description and the parameters.

eugenecsa commented on src/main/java/duke/application/MainWindow.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    /**
+     * Set the parser
```
Remember to have an empty line between the description and the parameters.

eugenecsa reviewed (commented):
Great work! The code is clear and well organized.

Do note that punctuation is required after each parameter description as per the module's Java convention.

Finally, you may consider adding more comments to the code, especially between blocks of code. This might make it easier to navigate the code in future.

HalphasX commented on src/main/java/duke/application/MainWindow.java:
```diff
@@ -0,0 +1,77 @@
+    private Parser parser;
+
+    /**
+     * Initialise settings, and greet user
```
A few of these Javadoc's headers for this method should have "s", such as initializes, greets, etc.

HalphasX commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    /**
+     * Parse the user input based on the command's possible parameters
```
This should have been parses instead of parse 😄

HalphasX reviewed (commented):
Great code quality! There is not much to comment about since it LGTM after fixing a few nits. All in all, good job!

clementkfj commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,59 @@
+     *
+     * @param input Full user input
+     * @param taskList The list of tasks
+     * @return The response
```
What response is it going to return?

clementkfj commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,40 @@
+     *
+     * @param input Full user input
+     * @param taskList The list of tasks
+     * @return The response
```
What response is it going to return?

clementkfj commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,25 @@
+     *
+     * @param input Full user input
+     * @param taskList The list of tasks
+     * @return The response
```
Noticed in a few classes but response differs for each classes so I thought maybe it would be good to show what kind of response it will have in the javadocs?

clementkfj commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,203 @@
+
+    /**
+     * Marks the specified task as done
+     * Also saves the taskList and queues a message to be printed
```
I noticed something and it's a pretty small and nitpicky thing but maybe standardise your statements to end with a fullstop?

clementkfj reviewed (commented):
Overall code looks tidy and neat! Small things that I commented were quite nitpicky and not a huge problem! Great job!

jonas-chow committed:
Fix formatting in comments

Javadocs comments are inconsistently formatted. Some have missing periods, missing capitalization, missing linebreaks, or poor grammar.

Standardising the format of these comments improves the code quality.

As a step towards such standardisation, let's ensure that all lines in javadocs comments start with a capital letter, and end with a period. There should also be a line break between the description and the first parameter. Furthermore, the first word of the description should be an active word. Doing so will provide a standard to write future javadocs comments.

jonas-chow committed:
Add assertions to multiple functions

Assertions are not being used in the code.

Using assertions will allow us to verify that each part of the code is working as intended on each change. And allow us to catch more bugs.

As a step towards this more automated bug catching, let's set up some assertions to ensure that bugs don't go unchecked.

jonas-chow committed:
Improve code quality

There are several portions of code with highly nested code or overly complicated code.

Reducing the nesting and complexity of the code improves the code quality and makes it more readable.

As a step towards better code quality, let's break down complicated expressions into multiple simple expressions, and limit the amount of nesting in each block of code.

jonas-chow committed:
Remove redundant import statement

jonas-chow committed:
Merge pull request #1 from jonas-chow/branch-A-Assertions

Add assertions to multiple functions

jonas-chow committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/task/TaskList.java

jonas-chow committed:
Merge pull request #2 from jonas-chow/branch-A-CodeQuality

Improve code quality

jonas-chow committed:
Add CI workflow

jonas-chow committed:
Remove actions that checked for IO redirection

jonas-chow committed:
Allow users to add their own aliases to commands

jonas-chow committed:
Merge branch 'branch-BCD-Extension'

jonas-chow committed:
Improve code quality

jonas-chow committed:
Add functionality to scroll through past commands

jonas-chow committed:
Make UI dark themed, add more tests

jonas-chow committed:
Add different text color for displaying exceptions

jonas-chow committed:
Update documentation

jonas-chow committed:
Fix style

jonas-chow committed:
Add table of contents for user guide
